projects/igniteui-angular/src/lib/grids/headers/grid-header.component.ts

```diff
@@ -233,6 +233,9 @@ export class IgxGridHeaderComponent implements DoCheck, OnDestroy {
             this.sortDirection + 1 > SortingDirection.Desc ? SortingDirection.Asc : SortingDirection.Desc
             : this.sortDirection + 1 > SortingDirection.Desc ? SortingDirection.None : this.sortDirection + 1;
         this.sortDirection = sortDir;
+        if (groupingExpr){
```

Comment:
Yes, that's what we intent to do, because sorting/grouping can also be changed programmatically at any point. But we discussed it only in one of the PRs.

Reply:
Yes you are right, subscribing to sorting in groupby area would be the appropriate approach.